### PR TITLE
Use `zig fmt` to fix the casting changes in the latest zig version

### PR DESCRIPTION
**Pull request by Tekkunsan**
New change from the latest zig version uses `@destFromSrc()` instead of `@srcToDest()` for casting

### CODE REVIEW

**Comment by floooh** [comment]
Ah ok, tests are expected to fail because the CI pipeline doesn't use the Zig dev version.

I'll try to test the change manually before merging the PR, might take a couple of days though until I get around it.

**Comment by Tekkunsan** [comment]
This Fixes #40 but `zig fmt` also formats the code, so a lot has change

**Comment by nurpax** [comment]
IMO it'd be great to always format the Zig sources in Sokol with `zig fmt`.  Otherwise it's kind of hard to make PRs against these files.  (I have `zig fmt` integrated into VSCode "save file" and I think many others have it like this too.)

File: src/examples/blend.zig [hunk 1]
```diff
@@ -53,29 +53,27 @@ export fn init() void {
     state.bg_pip = sg.makePipeline(pip_desc);
 
     // lot of pipeline objects for rendering the blended quads
-    pip_desc = .{
-        .shader = sg.makeShader(shd.quadShaderDesc(sg.queryBackend())),
-        .primitive_type = .TRIANGLE_STRIP,
-        .blend_color = .{ .r=1.0, .g=0.0, .b=0.0, .a=1.0 }
```
Comment (Manuzor) [inline src/examples/blend.zig:59]: With a trailing comma for the last argument, `zig fmt` will preserve multiple lines for `pip_desc` initialization. In other words, add a comma `,` after the closing brace `}` of `.blend_color = ...`.

```zig
.blend_color = .{ .r=1.0, .g=0.0, .b=0.0, .a=1.0 },
```

File: src/examples/blend.zig [hunk 2]
```diff
@@ -94,19 +92,15 @@ export fn frame() void {
...
+    var src: usize = 0;
+    while (src < NUM_BLEND_FACTORS) : (src += 1) {
+        var dst: usize = 0;
+        while (dst < NUM_BLEND_FACTORS) : (dst += 1) {
             // compute model-view-proj matrix
             const shift = NUM_BLEND_FACTORS / 2;
-            const t: vec3 = .{
-                .x = (@intToFloat(f32, dst) - shift) * 3.0,
-                .y = (@intToFloat(f32, src) - shift) * 2.2,
-                .z = 0.0
```
Comment (Manuzor) [inline src/examples/blend.zig:104]: Again, trailing comma would preserve the multiple lines here.

**Comment by Tekkunsan** [comment]
Tried to preserve the old formatting but I'm sure I'm missing a lot.

**Comment by nurpax** [comment]
Looks good to me at least.  That's the style of Zig that I see everywhere.  

I used to sweat over code formatting, like aligning tables & such, but the convenience of just hitting Save in VSCode and letting it run "zig fmt" to clean things up is so high.

**Comment by floooh** [comment]
Hmm, we should probably fix the code generation so that the output matches `zig fmt` (the Rust generator currently has the same problem (codegen doesn't match `cargo fmt`, and that's a bit of a hassle).

This can wait a bit though.

**Comment by nurpax** [comment]
> This is my main complaint with formatting tools, anything that's not properly aligned seems to trigger my OCPD ;)

Yup, that's been my biggest issue with auto-formatting tool.  At some point I just gave up though since auto-formatting seem to be increasingly prevalent (esp. with golang where it's basically required for any real work) and none of these tools seem to care much for this type of alignment.  I wouldn't go back to manual formatting anymore, "format on save" is just so convenient and you learn to rely on it a lot.

It's nice that at least line breaks can be controlled pretty well with added commas, as was used in this PR too.

**Comment by floooh** [comment]
...going to merge this now, and then apply the new bindgen with @nurpax's formatting fixes.

**Comment by floooh** [comment]
ah... when doing this I also need to fix the samples for the last round of breaking changes doh... (the 0.11.0 branch unfortunately wasn't updated yet...)

**Comment by floooh** [comment]
Ok, the zig-0.11.0 branch should now be in good shape, and also updated with the latest headers and example fixes from master.

**Pull request merged**